### PR TITLE
 Add '--version' flag and allow version and help flags when called as 'cargo-clippy'

### PR DESCRIPTION
**Pull request by Machtan**
This makes it easier to integrate 'cargo clippy' into other tools. 
(Or at least makes it possible to reason about the capabilities of the tool based on its version).

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -140,26 +142,42 @@ fn show_help() {
...
-        }
-
         let manifest_path_arg = std::env::args().skip(2).find(|val| val.starts_with("--manifest-path="));
 
-        let mut metadata = cargo::metadata(manifest_path_arg.as_ref().map(AsRef::as_ref)).expect("could not obtain cargo metadata");
+        let mut metadata = if let Ok(metadata) = cargo::metadata(manifest_path_arg.as_ref().map(AsRef::as_ref)) {
+            metadata
+        } else {
+            let _ = io::stderr().write_fmt(format_args!("error: Could not obtain cargo metadata."));
+            process::exit(101);
```
Comment (oli-obk) [inline src/main.rs]: we have a pretty random policy with `panic`s vs `process::exit`, so I guess this is fine. What's the reason for this change though?

Comment (oli-obk) [inline src/main.rs]: there's a million other cases where `cargo clippy` can panic. I think we should do a clean refactor of the main function to use `Result` based error handling and then have a single location that calls `process::exit` but that's for a different PR :)

File: src/main.rs [hunk 2]
```diff
@@ -140,26 +142,42 @@ fn show_help() {
     println!("{}", CARGO_CLIPPY_HELP);
 }
 
+#[allow(print_stdout)]
+fn show_version() {
+    println!("{}", env!("CARGO_PKG_VERSION"));
+}
+
+#[cfg_attr(feature = "cargo-clippy", allow(print_stdout))]
```
Comment (oli-obk) [inline src/main.rs]: why is this attribute added? And why only under the `cargo-clippy` feature?

**Comment by Machtan** [comment]
It was just an artifact from when I was trying to please the dogfood test. Removed now.

**Comment by oli-obk** [comment]
thanks!
